tests/api/v1/users/patch.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2016, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/api/v1/users/patch.js
+ */
+'use strict';
+
+const supertest = require('supertest');
+const api = supertest(require('../../../../index').app);
+const constants = require('../../../../api/v1/constants');
+const tu = require('../../../testUtils');
+const u = require('./utils');
+const path = '/v1/users';
+const expect = require('chai').expect;
+const jwtUtil = require('../../../../utils/jwtUtil');
+const Profile = tu.db.Profile;
+const User = tu.db.User;
+const Token = tu.db.Token;
+
+describe(`api: PATCH ${path}`, () => {
+  const ZERO = 0;
+  const ONE = 1;
+  const TWO = 2;
+  let profileOneId = '';
+  let profileTwoId = '';
+  const adminUser = require('../../../../config').db.adminUser;
+  const userOne = `${tu.namePrefix}test@refocus.com`;
+  const userTwo = `${tu.namePrefix}poop@refocus.com`;
+  const userThree = `${tu.namePrefix}quote@refocus.com`;
+  const tname = `${tu.namePrefix}Voldemort`;
+  const pname = `${tu.namePrefix}testProfile`;
+  const normalUserToken = jwtUtil.createToken(
+    userOne, userOne
+  );
+  const adminUserToken = jwtUtil.createToken(
+    adminUser.name, adminUser.name
+  );
+
+  before((done) => {
+    Profile.create({
+      name: pname + ONE,
+    })
+    .then((profile) => {
+      profileOneId = profile.id;
+      return Profile.create({
+        name: pname + TWO,
+      });
+    })
+    .then((profile) => {
+      profileTwoId = profile.id;
+      User.create({
+        profileId: profileOneId,
+        name: userOne,
+        email: userOne,
+        password: userOne,
+      });
+    }) // another normal user
+    .then(() => User.create({
+        profileId: profileTwoId,
+        name: userTwo,
+        email: userTwo,
+        password: userTwo,
+      })
+    ) // another normal user
+    .then(() => User.create({
+        profileId: profileTwoId,
+        name: userThree,
+        email: userThree,
+        password: userThree,
+      })
+    )
+    .then(() => done())
+    .catch(done);
+  });
+
+  after(u.forceDelete);
+
+  it('admin FORBIDDEN from changing their profileId', (done) => {
```

Comment:
@iamigo I think this is valid test case. OB Admin cannot change its own profile id. But non OB Admin CAN change their profile Id. Can you confirm?
@annyhe If above is true, we will need 2 tests that non OB user can change their own (and other non OB admin user) profile Ids